Clojure-Basics.md

```diff
@@ -0,0 +1,111 @@
+(def hello-world (fn [] (println our-string)))
+```
+`defn` is just a shortcut to help keep your code concise. `(defn ...)` and `(def ...
+(fn ...))` are the same in practice.
+
+## Parameters
+
+Well, that was nice, but it wasn't really exciting. Let's try a function with
+parameters. How about one that adds 3 numbers?
+```clojure
+(defn add [x y z] (+ x y z))
+(add 1 2 3)
+; => 6
+```
+...Hold on. `(+ x y z)`? That looks funny. Well, Lisps are written using "prefix
+notation", which means that the function always comes first. Since all
+mathematical operators in Lisp (`+ - * /`) are just functions, they also come
+before their arguments (in this case, `x y z`).
+
+You'll notice that our vector has some stuff in it now: `[x y z]`! Whenever
+a function has parameters, you define them in that vector next to the function's
+name.
+
+### Destructuring
+
+A great feature about arguments in Clojure is destructuring. It allows you to
+'pull' elements out of a list.
+```clojure
+(defn add [[x y] z] (+ x y z))
+(add [1 2] 3)
+; => 6
+```
+The arguments to that function are a collection (`[x y]`) and a number (`z`).
+We can use destructuring to pull the first and second elements out of the list,
+and call them `x` and `y`.
+
+### Functions with any number of parameters
+
+You can also define a function with an arbitrary number of arguments using `&`.
+```clojure
+(defn demonstrate-rest [first & rest]
+  (println first)
+  (println rest))
+(demonstrate-rest 1 "foo" ["bar" 22])
+; => 1
+;    ("foo" ["bar" 22])
+;    nil
+```
+As you can see, using `&` separated our function's arguments into one variable
+called `first` and a list of variables called `rest`. This means that our function
+could have any number of arguments!
+
+## Returning
+
+You may have noticed some odd things. Whenever we use `println`, it seems like
+`nil` is showing up in our output. Furthermore, our `add` function returns `6`,
+but we never told it to return anything.
+
+In Clojure, returns are 'implicit'. If you've used Ruby, you're probably
+familiar with this concept. Rather than telling our function to return something,
+it evaluates all the code inside its body, and returns the result. Our `add`
+function, for example, evaluates `(+ x y z)`, and then returns that result.
+
+The reason why our functions that use `println` output `nil` is because `println`
+evaluates to `nil`. (`nil` is like `null` or `None` -- it represents nothing.)
+
+| [:point_left: Previous](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure-Summary.md) | [:book: Home :book:](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure.md) | [Next :point_right:](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure-Conditionals.md)|
+|:---|:---:|----:|
+| [Summary](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure-Summary.md) | [Table of Contents](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure.md) | [Conditionals](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure-Conditionals.md)|
```

Comment:
~~`https://github.com/FreeCodeCamp/wiki/blob/master/Clojure.md`~~ links can just be the name of file without extension (`.md`).

For example:
```markdown
| [:point_left: Previous](Clojure-Summary) | [:book: Home :book:](Clojure) | [Next :point_right:](Clojure-Conditionals)|
|:---|:---:|----:|
| [Summary](Clojure-Summary) | [Table of Contents](Clojure) | [Conditionals](Clojure-Conditionals)|
```